test/Confluent.SchemaRegistry.Serdes.UnitTests/SerializeDeserialize.cs

```diff
@@ -147,6 +148,43 @@ public void ISpecificRecord()
             Assert.Equal(user.favorite_color, result.favorite_color);
             Assert.Equal(user.favorite_number, result.favorite_number);
         }
+        
+        [Fact]
+        public void Multiple_ISpecificRecords()
+        {
+            var serializer = new AvroSerializer<ISpecificRecord>(schemaRegistryClient);
+            var deserializerUser = new AvroDeserializer<User>(schemaRegistryClient);
+            var deserializerCar = new AvroDeserializer<Car>(schemaRegistryClient);
```

Comment:
The big API problem is on the deserializer side, which you haven't tackled. I assume you're reading bytes and have developed code on top of the consumer for this. that this is non-trivial is the reason we haven't done anything adding this capability yet - we do have plans for a higher level API that sits on top of the consumer which enables working with multiple schemas (allows functions to be registered to handle messages of different types). this would also incorporate additional capability (threads, etc). anyway, this is just a side note.